### DISCUSSION TITLE
Kubernetes self deployment

### DISCUSSION THREAD

**Discussion by mike12345567** [Budibase/budibase#984] [Ideas]
<p align="center">
<img src="https://miro.medium.com/max/800/1*WpKHLIDsJZgWKJe-SkOtcg.png" width="128">
</p>

Budibase now supports self hosting with the help of [Docker](https://www.docker.com/)/[Docker compose](https://docs.docker.com/compose/) but we want to keep making it easier to build your ideas with our tools. As part of this we plan to introduce more self hosting deployment options, one option for this is support for K8S (Kubernetes).

If you would be interested in a simple K8S deployment of Budibase this is the place to talk about it! We would love to know your use case for this and why you would prefer this over any other deployment option.

**Comment by Hokwang** [comment]
I need helm chart for budibase!

**Comment by aantn** [comment]
I'm still evaluating Budibase, but if I use it then this would be a MUST and I would be happy to contribute the Kubernetes deployment files. @mike12345567 have you done any work mapping out the effort here and what would be involved? Is this just a straightformward Docker Compose -> Kubernetes transition?

**Reply by shogunpurple** [reply]
Hi @aantn, we actually discussed this during the week and will be experimenting with this over the next few weeks. We will keep everyone up to date in this thread on the status of it. We should be able to use kompose to produce a basic kubectl file from our existing docker-compose config.

**Reply by aantn** [reply]
Cool, I'm happy to help out in any way. I develop an open source platform for Kubernetes, so I have the relevant experience.

**Comment by mjashanks** [comment]
K8s support has been released: https://github.com/Budibase/budibase/discussions/2508 @Hokwang @aantn

**Reply by datapolitical** [reply]
What's the biggest difference between deploying on k8s and docker?

Why would I choose one over the other, and which one is more stable?

**Reply by shogunpurple** [reply]
https://www.theserverside.com/blog/Coffee-Talk-Java-News-Stories-and-Opinions/What-is-Kubernetes-vs-Docker-Compose-How-these-DevOps-tools-compare

TLDR;
**Kubernetes** 
- larger scale deployments across 1 or more servers, only use if you know how to manage kubernetes
- more effort to set up and maintain, but more scalable for large deployments
- more metrics and insight out of the box with K8S monitoring tools

**Docker compose**
- single server deployments
- easier to manage and provision
- Nice integration with docker management tools like portainer

Both stable - if you are comfortable self hosting, managing updates and general environment maintenance. If this is not something you have ever done before, it's easier to use budibase cloud and have that all handled for you.
